src/watertap_contrib/reflo/analysis/case_studies/KBHDP/components/EC.py

```diff
@@ -40,9 +40,74 @@
 from watertap.costing import WaterTAPCosting
 import math
 
+__all__ = [
+    "build_ec",
+    "set_ec_operating_conditions",
+    "init_ec",
+    "add_ec_costing",
+    "add_ec_scaling",
+    "report_EC",
+    "print_EC_costing_breakdown",
+]
+
 
-def propagate_state(arc):
+def propagate_state(arc, detailed=True):
     _prop_state(arc)
+    if detailed:
+        print(f"Propogation of {arc.source.name} to {arc.destination.name} successful.")
+        arc.source.display()
+        print(arc.destination.name)
+        arc.destination.display()
+        print("\n")
+
+
+def _initialize(blk, verbose=False):
+    if verbose:
+        print("\n")
+        print(
+            f"{blk.name:<30s}{f'Degrees of Freedom at Initialization = {degrees_of_freedom(blk):<10.0f}'}"
+        )
+        print("\n")
+    try:
+        blk.initialize()
+    except:
+        print("----------------------------------\n")
+        print(f"Initialization of {blk.name} failed.")
+        print("\n----------------------------------\n")
+
+        blk.report()
+        print_infeasible_bounds(blk)
+        print_close_to_bounds(blk)
+        assert False
+
+
+def build_system():
+    """Function to create concrete model for individual unit model flowsheet"""
+    m = ConcreteModel()
+    m.db = REFLODatabase()
+
```

Comment:
FYI I think you will need a constraint here to calculate the conductivity from the TDS, otherwise it uses the default value of 1 mS/cm. You can see what I have in the Permian component file.